### ISSUE TITLE
Segfault when using UNIVERSAL::can + eval

### ISSUE THREAD

**Issue by atoomic** [rurban/perl-compiler#254]
ENV: perl 5.22 with debug enabled

perl
````
> perl -e 'Foo->UNIVERSAL::can("boogie"); print "ok\n" unless eval q/Foo->boogie(); 1/;'
ok
```

perlcc
```
$>perlcc -e 'Foo->UNIVERSAL::can("boogie"); print "ok\n" unless eval q/Foo->boogie(); 1/;'; ./a.out
zsh: segmentation fault  ./a.out
````

**Comment by rurban** [comment]
for us only Foo->can() is high prio. let's see

**Comment by toddr** [comment]
I just looked into this. Foo->UNIVERSAL::can while silly looking is not deprecated. So we should still investigate this down the road.

**Comment by toddr** [comment]
./t/testc.sh 254

**Comment by toddr** [comment]
This has been resolved by commit 55528ddb58f5a549115ac27455ed58d0c88c6edc which is strange cause I thought this was an overload issue? 55528ddb58f5a549115ac27455ed58d0c88c6edc
